Sources/BSON/ExtendedJSONEncoder.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+/// Facilitates the encoding of `Encodable` values into ExtendedJSON.
+public class ExtendedJSONEncoder {
+    internal enum extendedJSONMode {
+        case canonical
+        case relaxed
+    }
+
+    /// Initialize an `ExtendedJSONEncoder`.
+    public init() {
```

Comment:
This initializer will eventually be taking in options right? Should I add something as part of this PR to include that?

Reply:
discussed offline: yep, add options to this

Reply:
done